exercises/tuples-lists-dictionaries/challenge_073.py

```diff
@@ -0,0 +1,58 @@
+# TUPLES, LISTS AND DICTIONARIES
+
+# Challenge 073
+"""
+Ask the user to enter four of their favourite foods and store them in a dictionary 
+so that they are indexed with numbers starting from 1. Display the dictionary in full, 
+showing the index number and the item. Ask them which they want to get rid of and 
+remove it from the list. Sort the remaining data and display the dictionary.
+"""
+
+
+def get_user_input(message):
+    while True:
+        user_input = input(message)
+        if len(user_input) == 0:
+            print("Please enter a food name.")
+        return user_input
+
+
+def get_user_input_integer(message):
+    while True:
+        user_input = input(message)
+        if user_input.isdigit():
+            return int(user_input)
+        print("Oops! That was not a valid number. Try again...")
+
+
+def ask_for_user_foods():
+    food_dictionary = {}
+    food_dictionary[1] = get_user_input("What's your first food item? ")
```

Comment:
Can you think of a way of implementing this that doesn't require duplication or the addition of more lines if requirements change later?

Reply:
I have renamed the file to remove the hyphens, and added a loop to prompt for the food items: https://github.com/gaebar/python-by-example-exercises/blob/e8516180d930875f16e1d37ca279b1cee9dca04a/exercises/tuples_lists_dictionaries/challenge_073.py